l10n_eu_nace/migrations/16.0.1.0.0/migrate.py

```diff
@@ -0,0 +1,38 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    """Merge res.partner.nace with res.partner.industry
+    ============================================================================
+    The field mapping is the following:
+    res.partner.nace -> res.partner.industry
+    - name           -> name
+    - name + code    -> full_name
+    - parent_id      -> parent_id
+    - child_ids      -> child_ids
+    - active         -> active
+    """
+    # First create the new industries
+    env.cr.execute(
+        """
+        INSERT INTO res_partner_industry (name, full_name, active)
+        SELECT name, CONCAT(code, ' - ', name), active FROM res_partner_nace
```

Comment:
Probably this will help to construct a ({code - name}, id) mapping:

```suggestion
        SELECT name, CONCAT(code, ' - ', name), active FROM res_partner_nace
        RETURNING full_name, id
```

Then you can use it later to update relations and partners.